core/Decode.cpp

```diff
@@ -112,6 +123,20 @@ namespace olympia
         }
     }
 
+    // process vset settings being forward from execution pipe
+    // for set instructions that depend on register
+    void Decode::process_vset_(const InstPtr & inst)
+    {
+        VCSRs_.lmul = inst->getLMUL();
+        VCSRs_.sew = inst->getSEW();
+        VCSRs_.vl = inst->getVL();
+        VCSRs_.vta = inst->getVTA();
+        VCSRs_.vlmax = (Inst::VLEN / VCSRs_.sew) * VCSRs_.lmul;
+        waiting_on_vset_ = false;
```

Comment:
Might want to try only scheduling Decode if `waiting_on_vset_` was true.

Reply:
Addressed in [73ca9f4](https://github.com/riscv-software-src/riscv-perf-model/pull/167/commits/73ca9f4adb0bce0b8756a9b1ab3d88bf96c589bd)